controllers/propagator/placement_change_reconclier.go

```diff
@@ -0,0 +1,190 @@
+		//nolint:forcetypeassert
+		pb := obj.(*policiesv1.PlacementBinding)
+
+		log := log.WithValues("placementBindingName", pb.GetName(), "namespace", pb.GetNamespace())
+		log.V(2).Info("Reconcile request for a PlacementBinding")
+
+		return common.GetPoliciesInPlacementBinding(ctx, r.Client, pb)
+	}
+
+	// only reconcile when the pb contains a policy or a policyset as a subject
+	pbPredicateFuncs := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			//nolint:forcetypeassert
+			pbObjNew := e.ObjectNew.(*policiesv1.PlacementBinding)
+			//nolint:forcetypeassert
+			pbObjOld := e.ObjectOld.(*policiesv1.PlacementBinding)
+
+			return common.IsForPolicyOrPolicySet(pbObjNew) || common.IsForPolicyOrPolicySet(pbObjOld)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			//nolint:forcetypeassert
+			pbObj := e.Object.(*policiesv1.PlacementBinding)
+
+			return common.IsForPolicyOrPolicySet(pbObj)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			//nolint:forcetypeassert
+			pbObj := e.Object.(*policiesv1.PlacementBinding)
+
+			return common.IsForPolicyOrPolicySet(pbObj)
+		},
+	}
+
+	// placementRuleMapper maps from a PlacementRule to the PlacementBindings associated with it
+	placementRuleMapper := func(ctx context.Context, object client.Object) []reconcile.Request {
+		log := log.WithValues("placementRuleName", object.GetName(), "namespace", object.GetNamespace())
+		log.V(2).Info("Reconcile Request for PlacementRule")
+
+		pbList := &policiesv1.PlacementBindingList{}
+		lopts := &client.ListOptions{Namespace: object.GetNamespace()}
+
+		opts := client.MatchingFields{"placementRef.name": object.GetName()}
+		opts.ApplyToList(lopts)
+
+		if err := r.List(ctx, pbList, lopts); err != nil {
+			return nil
+		}
+
+		var result []reconcile.Request
+		// loop through pbs find the matching ones
+		for _, pb := range pbList.Items {
+			match := pb.PlacementRef.APIGroup == appsv1.SchemeGroupVersion.Group &&
+				pb.PlacementRef.Kind == "PlacementRule" &&
+				pb.PlacementRef.Name == object.GetName()
+
+			if match {
+				result = append(result, common.GetPoliciesInPlacementBinding(ctx, r.Client, &pb)...)
+			}
+		}
+
+		return result
+	}
+
+	// placementDecisionMapper maps from a PlacementDecision to the PlacementBindings associated with its Placement
+	placementDecisionMapper := func(ctx context.Context, object client.Object) []reconcile.Request {
+		log := log.WithValues("placementDecisionName", object.GetName(), "namespace", object.GetNamespace())
+		log.V(2).Info("Reconcile request for a placement decision")
+
+		// get the Placement name from the PlacementDecision
+		placementName := object.GetLabels()["cluster.open-cluster-management.io/placement"]
+		if placementName == "" {
+			return nil
+		}
+
+		pbList := &policiesv1.PlacementBindingList{}
+		lopts := &client.ListOptions{Namespace: object.GetNamespace()}
+
+		opts := client.MatchingFields{"placementRef.name": placementName}
```

Comment:
Similar comment to before. Are you able to filter on `placementRef.kind` as well?